sdk/provisioning/Azure.Provisioning/src/Expressions/BicepTypeMapping.cs

```diff
@@ -162,8 +162,12 @@ BicepExpression CompileNestedConstruct(ProvisionableConstruct construct)
     private static string GetEnumValue(Enum value)
     {
         Type type = value.GetType();
-        string name = Enum.GetName(type, value);
-        DataMemberAttribute? member = type.GetField(name)?.GetCustomAttribute<DataMemberAttribute>();
-        return member?.Name ?? value.ToString();
+        string? name = Enum.GetName(type, value);
```

Comment:
I was going to ask why nullability was leaking through here, but looks like `Azure.Provisioning` explicitly opts-in.  _([src](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/provisioning/Azure.Provisioning/src/Azure.Provisioning.csproj#L7))_